### PR TITLE
finished auth

### PR DESCRIPTION
**Pull request by medubin**
💯  @rdavidreid  🎱  @nhatho89 
### Goal

add auth to our application
### Testing

you should be able to create an account and log in via local
### Notes

Custom built
No devise

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -37,10 +37,16 @@ GEM
...
+    bcrypt (3.1.11)
+    better_errors (2.1.1)
+      coderay (>= 1.0.0)
+      erubis (>= 2.6.6)
+      rack (>= 0.9.0)
     binding_of_caller (0.7.2)
       debug_inspector (>= 0.0.1)
     builder (3.2.2)
     byebug (9.0.5)
+    coderay (1.1.0)
```
Comment (rdavidreid) [inline Gemfile.lock:49]: waht is this?

File: app/assets/javascripts/sessions.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (rdavidreid) [inline app/assets/javascripts/sessions.coffee]: delete

File: app/assets/javascripts/users.coffee [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
Comment (rdavidreid) [inline app/assets/javascripts/users.coffee]: delete

File: Gemfile.lock (hunk 1 above)
Comment (medubin) [reply Gemfile.lock:49]: This is supposed to be in the Gemfile.lock

File: app/assets/javascripts/sessions.coffee (hunk 2 above)
Comment (medubin) [reply app/assets/javascripts/sessions.coffee]: Done.

File: app/assets/javascripts/users.coffee (hunk 3 above)
Comment (medubin) [reply app/assets/javascripts/users.coffee]: Done.

**Comment by medubin** [comment]
Can I close this and push it to Master?

**Pull request merged**